components/lc709203f/lc709203f.c

```diff
@@ -0,0 +1,359 @@
+/*
+ * Copyright (c) 2022 Jose Manuel Perez <user@your.dom.ain>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/*
+ * this file is supposed to conform the code style.
+ */
+
+#include <freertos/FreeRTOS.h>
+#include <esp_idf_lib_helpers.h>
+#include <esp_err.h>
+#include <esp_log.h>
+
+#include "lc709203f.h"
+
+#define LC709203F_I2C_ADDR        0x0B   ///< LC709203F default i2c address
+#define LC709003F_I2C_MAX_FREQ_HZ 400000 ///< 400kHz
+
+#define LC709203F_REG_BEFORE_RSOC       0x04 ///< Initialize before RSOC
+#define LC709203F_REG_THERMISTOR_B      0x06 ///< Read/write thermistor B
+#define LC709203F_REG_INITIAL_RSOC      0x07 ///< Initialize RSOC calculation
+#define LC709203F_REG_CELL_TEMPERATURE  0x08 ///< Read/write batt temperature
+#define LC709203F_REG_CELL_VOLTAGE      0x09 ///< Read batt voltage
+#define LC709203F_REG_CURRENT_DIRECTION 0x0A ///< Read/write current direction
+#define LC709203F_REG_APA               0x0B ///< Read/write Adjustment Pack Application
+#define LC709203F_REG_APT               0x0C ///< Read/write Adjustment Pack Thermistor
+#define LC709203F_REG_RSOC              0x0D ///< Read state of charge
+#define LC709203F_REG_CELL_ITE          0x0F ///< Read batt indicator to empty
+#define LC709203F_REG_IC_VERSION        0x11 ///< Read IC version
+#define LC709203F_REG_CHANGE_PARAMETER  0x12 ///< Set the battery profile
+#define LC709203F_REG_ALARM_LOW_RSOC    0x13 ///< Alarm on percent threshold
+#define LC709203F_REG_ALARM_LOW_VOLTAGE 0x14 ///< Alarm on voltage threshold
+#define LC709203F_REG_IC_POWER_MODE     0x15 ///< Sets sleep/power mode
+#define LC709203F_REG_STATUS_BIT        0x16 ///< Temperature obtaining method
+#define LC709203F_REG_NUM_PARAMETER     0x1A ///< Batt profile code
```

Comment:
put these resister macros to the header if there is not a specific reason to hide them. users might need resister addresses.

Reply:
Ok. 

There is getter&setter version for each R/W register and only setter or getter for W or R. I can't imagine where user would need this specific value but it's ok, I will move them.